storage/s3_storage.py

```diff
@@ -42,16 +43,16 @@ def save_to_filename(self, file_path: str) -> None:
         transfer = boto3.s3.transfer.S3Transfer(client)
         try:
             transfer.download_file(self._bucket, self._keyname, file_path)
-        except IOError:
-            raise NotFoundError("No Files Found")
+        except ClientError:
+            raise NotFoundError("No File Found")
```

Comment:
"[`ClientError`] is a general exception when an error response is provided by an AWS service to your Boto3 client’s request." ([boto3 docs](https://boto3.amazonaws.com/v1/documentation/api/latest/guide/error-handling.html#catching-botocore-exceptions)) This means a `ClientError` can be raised for **any** 4xx-level error. We should only raise a `NotFoundError` (and chain the original exception) when the exception is for a missing file. Otherwise, we should re-raise the original exception.

You will need to inspect the exception object to determine if the error was caused by a missing file. I suspect you can detect it by there being a 404 code somewhere inside the exception, but you will need to verify that is the case.